lib/OpenQA/WebAPI.pm

```diff
@@ -29,6 +29,7 @@ use DateTime;
 use Cwd qw/abs_path/;
 use File::Path qw/make_path/;
 use OpenQA::Worker::Common qw/ASSET_DIR/;
+use Scalar::Util 'weaken';
```

Comment:
Why on `$tx`? Its reference counter is again decreased once the header is added and hook goes out of scope. Isn't it? Am I missing something? We are not bringing `$tx` from "outside" there.